src/library/baseexternallibraryfeature.cpp

```diff
@@ -28,27 +28,48 @@ BaseExternalLibraryFeature::BaseExternalLibraryFeature(QObject* pParent,
 }
 
 BaseExternalLibraryFeature::~BaseExternalLibraryFeature() {
+    qDebug() << "";
+    qDebug() << "   ~BaseExternalLibraryFeature";
     delete m_pAddToAutoDJAction;
+    qDebug() << "     deleted m_pAddToAutoDJAction";
     delete m_pAddToAutoDJTopAction;
+    qDebug() << "     deleted m_pAddToAutoDJTopAction";
     delete m_pImportAsMixxxPlaylistAction;
+    qDebug() << "     deleted m_pImportAsMixxxPlaylistAction";
+    delete m_pMenu;
+    qDebug() << "     deleted m_pMenu";
+    qDebug() << "";
 }
 
 void BaseExternalLibraryFeature::onRightClick(const QPoint& globalPos) {
+    qDebug() << "   BaseExternalLibraryFeature::onRightClick()";
     Q_UNUSED(globalPos);
     m_lastRightClickedIndex = QModelIndex();
 }
 
-void BaseExternalLibraryFeature::onRightClickChild(const QPoint& globalPos, QModelIndex index) {
+void BaseExternalLibraryFeature::onRightClickChild(const QPoint& globalPos, QModelIndex index,
+        WLibrarySidebar* pSidebarWidget) {
+    qDebug() << "     BaseExternalLibraryFeature::onRightClickChild(" << pSidebarWidget << ")";
     //Save the model index so we can get it in the action slots...
     m_lastRightClickedIndex = index;
 
-    //Create the right-click menu
-    QMenu menu;
-    menu.addAction(m_pAddToAutoDJAction);
-    menu.addAction(m_pAddToAutoDJTopAction);
-    menu.addSeparator();
-    menu.addAction(m_pImportAsMixxxPlaylistAction);
-    menu.exec(globalPos);
+    if (m_pMenu != nullptr) {
+        qDebug() << "       m_pMenu != nullptr";
+//        delete m_pMenu;
```

Comment:
Here, it crashes with Banshee and Rhythmbox menu: I opened the Banshee menu, reloaded the skin, tried to open the Banshee menu again, crash. The same procedure works well for regular library features, why does it crash for here?
I can see that the sidebar widget pointer is properly passed to BaseExternalLibraryFeature, also after skin reload, and the QMenu is deleted and created before skin reload.

Any help si welcome.
Meanwhile I'll look into `unique_ptr` and try to simply overwrite `m_pMenu`

Reply:
You can safely delete a nullptr. So this cannot be the issue and the check can removed. I guess the m_pMenu; was already deleted and not set to nullptr.
If you want to continue the manual route, check if no copies of m_qMenu are deleted and that it is set to nullptr after delete.

Reply:
Thanks.
I also came across `m_pMenu->setAttribute(Qt::WA_DeleteOnClose);` which seems to do the trick. I can remove the checks and simplify the function. Still need to test this thoroughly for rightClick and rightClickChild on features

Reply:
Ouuuh damn,... This is just a leftover from when I created the menu the moment the sidebar widget was registered in each feature.
:upside_down_face:
I'll revert all the menu functions to the simple way they used before.
Thanks anyway! I learned a lot.